packages/data/src/settings/withSettings.js

```diff
@@ -0,0 +1,41 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { partial } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from './constants';
+
+export const withSettings = group =>
```

Comment:
The HOC is a good idea.  However I would create a `useSettings` custom hook with the necessary logic so it can be used directly in functional components vs wrapping with the hook (which keep the tree levels more flat).  Then you can still have a HOC for places where you want to avoid refactoring a `React.Component` but the hoc would just use the new hook.

Also, as mentioned earlier, it's an anti-pattern to pass the actual selector through to components because then they aren't subscribed to state changes.  One change you could make here to make it easier for consumers is have the custom hook (and the new hoc suggested earlier) receive an optional set of arguments for what specific setting changes to select (like what you did for the curried function using `group` here.